src/main/java/rx/internal/operators/OperatorConcat.java

```diff
@@ -179,6 +187,14 @@ void subscribeNext() {
                 }
             }
         }
+        
+        void produced(long c) {
+            arbiter.produced(c);
+            AtomicLong requestedField = requested;
+            if (requestedField.get() != Long.MAX_VALUE) {
+                requestedField.addAndGet(-c);
```

Comment:
In normal use, such change never happens; consumers either start with MAX_VALUE or some prefetch value, plus you have to be very precise to hit the narrow window between the two lines.